src/validation.h

```diff
@@ -142,15 +142,16 @@ void StartScriptCheckWorkerThreads(int threads_num);
 /** Stop all of the script checking worker threads */
 void StopScriptCheckWorkerThreads();
 /**
- * Return transaction from the block at block_index.
- * If block_index is not provided, fall back to mempool.
- * If mempool is not provided or the tx couldn't be found in mempool, fall back to g_txindex.
+ * Return transaction with a given hash.
+ * If mempool is provided and block_index is not provided, check it first for the tx.
+ * If -txindex is available, check it next for the tx.
+ * Finally, if block_index is provided, check for tx by reading entire block from disk.
  *
  * @param[in]  block_index     The block to read from disk, or nullptr
- * @param[in]  mempool         If block_index is not provided, look in the mempool, if provided
+ * @param[in]  mempool         If provided, check mempool for tx
```

Comment:
```suggestion
 * @param[in]  mempool         If provided and block_index is not, check mempool for tx
```

Reply:
I'd suggest just replacing this with `Optional pointer to the node's mempool` to avoid duplicating what's already documented above.